src/yz_index.erl

```diff
@@ -338,3 +374,45 @@ index_dir(Name) ->
 make_info(SchemaName, NVal) ->
     #index_info{n_val=NVal,
                 schema_name=SchemaName}.
+
+-spec sync_index(pid(), index_name(), timeout()) -> ok | timeout |
+                                                   {core_error, binary()}.
+sync_index(Pid, IndexName, Timeout) ->
+    process_flag(trap_exit, true),
+    {ok, Ring} = riak_core_ring_manager:get_my_ring(),
+    Nodes = riak_core_ring:all_members(Ring),
+    WaitPid = spawn_link(?MODULE, wait_for_index, [self(), IndexName, Nodes]),
+    register(wait_for_index, WaitPid),
+    receive
+        {_From, ok} ->
+            Pid ! ok;
+        {'EXIT', _Pid, _Reason} ->
+            sync_index(Pid, IndexName, Timeout)
+    after Timeout ->
+            exit(whereis(wait_for_index), kill),
+
+            %% Check if initFailure occurred
+            {ok, _, S} = yz_solr:core(status, [{wt,json},{core, IndexName}]),
+            case ?INITFAILURES(IndexName, S) of
+                [] ->
+                    lager:warning("Index ~s not created within ~p ms timeout",
+                                  [IndexName, Timeout]),
+                    Pid ! timeout;
+                Error ->
+                    lager:warning("Solr Core Error on index ~s: ~p",
+                                  [IndexName, Error]),
+                    Pid ! {core_error, Error}
+            end
+    end.
+
+-spec wait_for_index(pid(), index_name(), [Node :: term()]) -> {pid(), ok}.
+wait_for_index(Pid, IndexName, Nodes) ->
+    Results =  riak_core_util:multi_rpc_ann(Nodes, ?MODULE, exists, [IndexName]),
+    case lists:filter(fun({_, Res}) -> Res =/= true end, Results) of
+        [] ->
+            Pid ! {self(), ok};
+        Rest ->
+            Nodes = [Node || {Node, _} <- Rest],
+            lager:debug("Index does not yet exist on Nodes ~p", [Nodes]),
```

Comment:
Another logging nitpick: Can this either be removed (how hot is this loop?) or be rewritten to give a little more context to what it means in a logfile? E.g. "Polled status of solr index ~s; does not yet exist on nodes ~p"